include/mrdox/Metadata/Template.hpp

```diff
@@ -14,12 +14,183 @@
 #define MRDOX_METADATA_TEMPLATE_HPP
 
 #include <mrdox/Platform.hpp>
-#include <mrdox/Metadata/TemplateArg.hpp>
-#include <mrdox/Metadata/TemplateParam.hpp>
+#include <mrdox/Metadata/Type.hpp>
+#include <string>
 
 namespace clang {
 namespace mrdox {
 
+enum class TParamKind : int
+{
+    None = 0, // for bitstream
```

Comment:
is "None" actually used or is this an invalid state?

Reply:
It's an intermediate state used when building when building the `TParam` (during AST traversal & deserialization)

Reply:
is it ever valid for a `Generator` to see `None` for a TParamKind ?

Reply:
If None is invalid then it must not be in the enumeration